File: keemsebin/5주차/5397.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];

rl.on("line", function (line) {
  input.push(line);
}).on("close", function () {
  let N = Number(input[0]);

  for (let i = 1; i <= N; i++) {
    const items = input[i];
    let leftStack = [];
    let rightStack = [];

    for (let char of items) {
      if (char === "<") {
        if (leftStack.length > 0) {
          rightStack.push(leftStack.pop());
        }
      } else if (char === ">") {
        if (rightStack.length > 0) {
          leftStack.push(rightStack.pop());
        }
      } else if (char === "-") {
        if (leftStack.length > 0) {
          leftStack.pop();
        }
      } else {
        leftStack.push(char);
      }
    }
    console.log(leftStack.join("") + rightStack.reverse().join(""));
  }
});

File: k-kbk/5주차/5397.py
import sys

input = sys.stdin.readline

n = int(input())

for i in range(n):
    line = list(input().strip())
    left = []
    right = []

    for j in line:
        if j == "<":
            if left:
                right.append(left.pop())
        elif j == ">":
            if right:
                left.append(right.pop())
        elif j == "-":
            if left:
                left.pop()
        else:
            left.append(j)

    left.extend(reversed(right))

    print("".join(left))
